#! /bin/sh /usr/share/dpatch/dpatch-run
## dhcp-3.1.0-ldap-fix-memory-leak.dpatch by  <apollock@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from Petter Reinholdtsen <pere@hungry.com> for #531050

@DPATCH@
diff -urNad dhcp3-3.1.3~/server/ldap.c dhcp3-3.1.3/server/ldap.c
--- dhcp3-3.1.3~/server/ldap.c	2010-01-08 17:16:30.205275093 -0800
+++ dhcp3-3.1.3/server/ldap.c	2010-01-08 17:17:22.985280336 -0800
@@ -269,6 +269,7 @@
           x_strncat (cfile->inbuf, tempbv[i]->bv_val, LDAP_BUFFER_SIZE);
           x_strncat (cfile->inbuf, ";\n", LDAP_BUFFER_SIZE);
         }
+        ldap_value_free_len (tempbv);
     }
 
   item->close_brace = 1;
